### PR TITLE
Changelog/add changelog check

### PR DESCRIPTION
**Pull request by breathingdust**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Adds a check to add a comment, and fail a CI check if a changelog entry is missing and a label excluding that PR from the check is not present. This also adds that label to the infrastructure project.

### CODE REVIEW

File: .github/workflows/changelog.yml
Comment (bflad) [inline .github/workflows/changelog.yml:55]: It looks like the link was updated, but not the description for it:

```suggestion
            Please note that the `CHANGELOG.md` file contents are generated automatically. This is to prevent pull request merge conflicts, especially for contributions which may not be merged immediately. Please see the [Contributing Guide](https://github.com/hashicorp/terraform-provider-aws/blob/main/docs/contributing/pullrequest-submission-and-lifecycle.md#changelog-process) for additional information on adding CHANGELOG entries.
```

File: .github/workflows/pull_requests.yml
Comment (bflad) [inline .github/workflows/pull_requests.yml:43]: I believe `dorny/paths-filter` can be used to replace this lookup logic, see also `changelog.yml` 👍

File: docs/contributing/pullrequest-submission-and-lifecycle.md
Comment (bflad) [inline docs/contributing/pullrequest-submission-and-lifecycle.md:222]: I do not believe contributors will be able to do this themselves, so the PR check will fail until a maintainer does it or the contributor (potentially extraneously) adds a file. I'm not sure if we want to mention that here as something that they should ask for or that the maintainers will ignore the failed check if the changelog entry is unnecessary.

**Review by bflad — COMMENTED** [review]
I also have some concerns about turning this on without filtering for pull request provider code modifications (e.g. only comment on `aws/*` and ignore `*_test.go`).

**Review by bflad — APPROVED** [review]
This looks good, only left a suggestion there for the comment itself. 👍

File: .github/workflows/changelog-check.yml
Comment (anGie44) [inline .github/workflows/changelog-check.yml:7]: LGTM 👍 just curious, if all PRs will be opened w/o a changelog file b/c we don't initially know the PR#, we'll expect this comment to be on all new PRs? or do we expect enough delay time for a user to push up the changelog before the job gets triggered? i remember reading around about accepting changelog's w/o #'s but can't find if that's in place or will be

Comment (anGie44) [reply .github/workflows/changelog-check.yml:7]: cool, def not too much, serves as a good reminder 👍

**Review by anGie44 — APPROVED** [review]

**Comment by breathingdust** [comment]
Uncomfortable with the comment and it appearing on every PR, so closing this out for now. We may revisit.

**Pull request closed without merging**
